### ISSUE TITLE
Path

### ISSUE THREAD

**Issue by Yonaskassa** [gnea/grbl#1210]
Hi Dear,
![Uploading 20230428_153604.jpg…]()
As you have seen a design on UGS the path is not random moving, Cnc moves unnecessary fwd and bwd movment. How to make the path moving from the nearest to other side.
Sorry for my english.

**Comment by breiler** [comment]
The picture wasn't uploaded properly

**Comment by Yonaskassa** [comment]
![20230428_153604](https://user-images.githubusercontent.com/131940631/235285001-a38d3e92-830e-433a-9cd5-d83c1a3529b5.jpg)

**Comment by breiler** [comment]
This is not an issue with GRBL, you would be better served by using the UGS discussions: https://github.com/winder/Universal-G-Code-Sender

In the latest version of UGS I removed the automatic ordering of objects as a request from another user.

Each geometric object will be processed in the order they are listed in the entities list. However if you have one complex geometric path object with a lot of holes the ordering of how each hole is processed could probably be optimized. Did you import an SVG, could you attach a sample file?

**Comment by Yonaskassa** [comment]
Sure, Here is attached sample .svg file.
Regards!
![Doors](https://user-images.githubusercontent.com/131940631/235307766-c2ba508d-280e-43b0-8c73-15136c2c8603.svg)

**Comment by breiler** [comment]
Yeah, this should be more optimized. I have created this issue in UGS: https://github.com/winder/Universal-G-Code-Sender/issues/2209

**Comment by breiler** [comment]
There is a new nightly build available that should optimize this a bit:
https://github.com/winder/Universal-G-Code-Sender/releases/tag/nightly
